source/llassetgen/include/llassetgen/Atlas.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <llassetgen/DistanceTransform.h>
+#include <llassetgen/Image.h>
+#include <llassetgen/Packing.h>
+
+namespace llassetgen {
+    namespace internal {
+        template <class Iter>
+        constexpr int checkImageIteratorType() {
+            using IterTraits = typename std::iterator_traits<Iter>;
+            using IterRefType = typename IterTraits::reference;
+            using IterCategory = typename IterTraits::iterator_category;
+            static_assert(std::is_assignable<Image, IterRefType>::value, "Input elements must be assignable to Image");
+            static_assert(std::is_base_of<std::input_iterator_tag, IterCategory>::value,
+                          "Input iterator must be an InputIterator");
+            return 0;
+        }
+    }
+
+    template <class ImageIter>
+    Image fontAtlas(ImageIter imgBegin, ImageIter imgEnd, Packing packing, uint8_t bitDepth = 1) {
+        internal::checkImageIteratorType<ImageIter>();
+        assert(std::distance(imgBegin, imgEnd) == static_cast<ptrdiff_t>(packing.rects.size()));
```

Comment:
If we want be completely generic we should use `std::iterator_traits<ImageIter>::difference_type` (which is what `std::distance` returns) instead of `ptrdiff_t`. Same goes for `distanceFieldAtlas`.

Reply:
Might as well, if your type name is shorter than 40 characters you ain't writing no real C++ anyway :grin: